trunk/src/protocol/srs_protocol_http_stack.cpp

```diff
@@ -691,14 +691,12 @@ srs_error_t SrsHttpServeMux::handle(std::string pattern, ISrsHttpHandler* handle
     srs_assert(handler);
     
     if (pattern.empty()) {
-        srs_freep(handler);
```

Comment:
The `ISrsHttpHander` memory release still not perfect.
For a normal process, the `ISrsHttpHandler` is created in:
https://github.com/ossrs/srs/blob/e3d74fb04504cce64f78119627efe3aae825dda3/trunk/src/app/srs_app_http_stream.cpp#L991
and released in 
https://github.com/ossrs/srs/blob/e3d74fb04504cce64f78119627efe3aae825dda3/trunk/src/protocol/srs_protocol_http_stack.cpp#L624-L627

I would think if it's possible to manager it in same place, `SrsLiveEntry` in this case. I see the auto release in `SrsHttpStreamServer::http_unmount(SrsRequest* r)`, it seems freed twice in `SrsHttpMuxEntry` destructor and `
SrsHttpStreamServer::http_unmount`

And For exception cases, like here, `handler` not freed here, but this exception not handled well.
https://github.com/ossrs/srs/blob/e3d74fb04504cce64f78119627efe3aae825dda3/trunk/src/app/srs_app_http_stream.cpp#L1008-L1010

Reply:
First, in http_unmount, won't free it twice, please check it carefully.

Second, if mux.handle failed, it won't free the stream, but the http_unmount will do.

So I think it's ok.